addons/portal/controllers/thread.py

```diff
@@ -0,0 +1,29 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.tools import html2plaintext
+from odoo.http import request
+from odoo.addons.mail.controllers import thread
+
+
+class ThreadController(thread.ThreadController):
+
+    @http.route()
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
+        message = super().mail_message_post(thread_model, thread_id, post_data, context, **kwargs)
+        message_data = message.get("Message")
+        if (message_data := message_data and message_data[0]) and request.env[
+            "res.partner"
+        ]._get_portal_partner_from_context():
+            message = {
+                "default_message": html2plaintext(message_data.get("body")),
+                "default_message_id": message_data.get("id"),
+                "default_attachment_ids": message_data.get("attachments"),
+            }
+        return message
```

Comment:
Can we do without this override of `mail_message_post`?

The idea of `_get_thread_with_access` and `_prepare_post_data` was that we would only need to override them very specifically to plug stuff into the base code.

I don't understand why we would need `default_` rather than just passing the correct params to the correct methods.